tools/profile_endpoints.py

```diff
@@ -0,0 +1,315 @@
+        )
+    else:
+        print(
+            f"Request failed with status code {response.status_code}: {response.text}"
+        )
+
+
+# HTTP Request Functions
+def make_get_request(endpoint_url, token):
+    """
+    Makes a GET request to the specified endpoint with the given access token.
+
+    Args:
+        endpoint_url (str): The endpoint URL to request.
+        token (str): The access token for authentication.
+
+    Returns:
+        float: The time taken for the request in seconds.
+    """
+    url = f"{DATAREPO_URL}/api/repository/v1/datasets/{UUID}/snapshotBuilder/{endpoint_url}"
+    headers = {"Authorization": f"Bearer {token}"}
+
+    try:
+        start_time = time.time()
+        response = requests.get(url, headers=headers)
+        end_time = time.time()
+
+        handle_response(response)
+        return end_time - start_time
+    except requests.exceptions.RequestException as e:
+        print(f"An error occurred during the request: {e}")
+        return None
+
+
+def make_post_request(endpoint_url, token, body):
+    """
+    Makes a POST request to the specified endpoint with the given access token and request body.
+
+    Args:
+        endpoint_url (str): The endpoint URL to request.
+        token (str): The access token for authentication.
+        body (dict): The request body to send in the POST request.
+
+    Returns:
+        tuple: A tuple containing the response object and the time taken for the request in seconds.
+    """
+    url = f"{DATAREPO_URL}/api/repository/v1/datasets/{UUID}/snapshotBuilder/{endpoint_url}"
+    headers = {
+        "Authorization": f"Bearer {token}",
+        "Content-Type": "application/json",  # Indicating the request body is in JSON format
+    }
+
+    try:
+        start_time = time.time()
+        response = requests.post(url, headers=headers, json=body)
+        end_time = time.time()
+        handle_response(response)
+        return end_time - start_time
+
+    except requests.exceptions.RequestException as e:
+        print(f"An error occurred during the request: {e}")
+        return None
+
+
+# Endpoint Profiling Functions
+def profile_get_snapshot_builder_count(token):
+    """
+    Profiles the getSnapshotBuilderCount endpoint and returns the time taken.
+
+    Args:
+        token (str): The access token for authentication.
+
+    Returns:
+        str: A string indicating the average time taken for the request.
+    """
+    print("Profiling getSnapshotBuilderCount endpoint")
+
+    # Define the bodies for the POST request
+    bodys = [
+        {
+            "cohorts": [
+                {
+                    "name": "",
+                    "criteriaGroups": [
+                        {
+                            "name": "Group 1",
+                            "mustMeet": True,
+                            "meetAll": False,
```

Comment:
I think there should be at least one test case with `"meetAll": True`. Also maybe a case with multiple criteria groups? Not sure how different the API performance is for different types of cohorts.